engine/table/src/main/java/io/deephaven/engine/table/impl/select/FilterSelectColumn.java

```diff
@@ -0,0 +1,269 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl.select;
+
+import io.deephaven.chunk.ChunkType;
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.WritableObjectChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.exceptions.UncheckedTableException;
+import io.deephaven.engine.rowset.*;
+import io.deephaven.engine.table.*;
+import io.deephaven.engine.table.impl.BaseTable;
+import io.deephaven.engine.table.impl.MatchPair;
+import io.deephaven.engine.table.impl.QueryCompilerRequestProcessor;
+import io.deephaven.engine.table.impl.QueryTable;
+import io.deephaven.engine.table.impl.sources.InMemoryColumnSource;
+import io.deephaven.engine.table.impl.sources.SparseArrayColumnSource;
+import io.deephaven.engine.table.impl.sources.ViewColumnSource;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The FilterSelectColumn wraps a {@link io.deephaven.api.filter.Filter} and producing a column of true or false Boolean
+ * values described by the filter.
+ *
+ * <p>
+ * This column is appropriate as an argument to a {@link Table#view(Collection)} or {@link Table#updateView(String...)},
+ * lazily evaluating the equivalent of a {@link Table#wouldMatch(String...)} operation. Although select and update can
+ * also use Filters, the wouldMatch provides a more efficient path for realized results as it stores and updates only a
+ * {@link RowSet}. The FilterSelectColumn can only evaluate the filter one chunk at a time, and must write to an
+ * in-memory Boolean column source.
+ * </p>
+ */
+class FilterSelectColumn implements SelectColumn {
+
+    // We don't actually care to do anything, but cannot return null
+    private static final Formula.FillContext FILL_CONTEXT_INSTANCE = new Formula.FillContext() {};
+
+    @NotNull
+    private final String destName;
+    @NotNull
+    private final WhereFilter filter;
+
+    private RowSet rowSet;
+    private Table tableToFilter;
+
+    /**
+     * Create a FilterSelectColumn with the given name and {@link WhereFilter}.
+     *
+     * @param destName the name of the result column
+     * @param filter the filter that is evaluated to true or false for each row of the table
+     * @return a new FilterSelectColumn representing the provided filter.
+     */
+    static FilterSelectColumn of(@NotNull final String destName, @NotNull final WhereFilter filter) {
+        return new FilterSelectColumn(destName, filter);
+    }
```

Comment:
I'd like us to do the transform from `api.Filter` to `WhereFilter` in this method.

If DHE needs to be able to construct this with a `WhereFilter` directly, we can have an additional method.